### PR TITLE
Ignore AllenNLP test from Mac-CI

### PR DESCRIPTION
**Pull request by cross32768**
## Motivation
AllenNLP test of Mac-CI sometimes fails (https://github.com/optuna/optuna/actions/runs/4552047524/jobs/8046281877 ).

Since AllenNLP's development is finished, we do not consider it necessary to fix the test and simply ignore it.

## Description of the changes
Ignore test relating allennlp from mac-ci

### CODE REVIEW

File: .github/workflows/mac-tests.yml [hunk 1]
```diff
@@ -121,7 +121,7 @@ jobs:
 
     - name: Tests
       run: |
-        pytest tests -m "integration"
+        pytest tests -m "integration" --ignore-glob="allennlp"
```
Comment (toshihikoyanase) [inline .github/workflows/mac-tests.yml:124]: How about following the way to ignore tests in `tests-integration.yml`?

https://github.com/optuna/optuna/blob/01aa6d9733269aa58e0ae8931a1bd28b662d7942/.github/workflows/tests-integration.yml#LL121C10-L121C60

```suggestion
        pytest tests -m "integration" --ignore tests/integration_tests/allennlp_tests
```

**Review by toshihikoyanase — COMMENTED** [review]
Thanks for your PR. I have a suggestion.

File: .github/workflows/mac-tests.yml (hunk 1 above)
Comment (cross32768) [reply .github/workflows/mac-tests.yml]: Thank you. That looks better and I fix it.

Comment (toshihikoyanase) [reply .github/workflows/mac-tests.yml]: Thanks!

**Review by toshihikoyanase — APPROVED** [review]
I confirmed that AllenNLP tests were skipped as expected. LGTM!

`master`
```console
tests/integration_tests/test_xgboost.py ...                              [ 38%]
tests/integration_tests/allennlp_tests/test_allennlp.py ................ [ 40%]
........F..                                                              [ 41%]
tests/integration_tests/lightgbm_tuner_tests/test_alias.py ............. [ 42%]
```

https://github.com/optuna/optuna/actions/runs/4552047524/jobs/8046281877

This PR:
```console
tests/integration_tests/test_xgboost.py ...                              [ 39%] 
tests/integration_tests/lightgbm_tuner_tests/test_alias.py ............. [ 41%]
```

https://github.com/optuna/optuna/actions/runs/4571930558/jobs/8070672174?pr=4561

**Review by gen740 — APPROVED** [review]
LGTM!

**Pull request merged**